src/pages/campaigns/NewCampaign.tsx

```diff
@@ -0,0 +1,250 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import * as z from "zod";
+import { format } from "date-fns";
+import { Type, FileText, Send, Calendar as CalendarIcon } from "lucide-react";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { Card, CardHeader, CardTitle, CardContent } from "@/components/ui/card";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import { useEffect, useState } from "react";
+import MDXComponent from "@/components/NewsletterComponents/MDXComponent";
+import Sidebar from "@/components/Sidebar";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { Calendar } from "@/components/ui/calendar";
+import { cn } from "@/lib/utils";
+import { useRecoilState } from "recoil";
+import { markdownState } from "@/atoms/atoms";
+import { marked } from "marked";
+const formSchema = z.object({
```

Comment:
Title and Blurb needs additional Regex checks within Zod. Currently, this is vulnerable to SQL-injection attacks. The regex checks are as follows:
1. a-z
2. A-Z
3. 0-9
4. Symbols = {",", ":", "?", "(", ")", "[", "]", "{", "}", "!", "&"}
5. Single-white-space character

The checking should not be strict as to make every-character mandatory but should be strict enough to make sure that if there are any unintended characters then they are not accepted. 

Also, the forms should not accept / send any preceding or trailing whitespaces.

Reply:
Done

Reply:
Looks good.